update/sync/syncHeadersByHash.go

```diff
@@ -84,24 +98,42 @@ func (m *syncHeadersByHash) SyncMissingHeadersByHash(shardIDs []uint32, headersH
 
 	for {
 		requestedHdrs := 0
+		requestedProofs := 0
 
 		m.mutMissingHdrs.Lock()
 		m.stopSyncing = false
 		for hash, shardId := range mapHashesToRequest {
+			hasProof := m.hasProof(shardId, []byte(hash))
```

Comment:
I think we need to extract into a function from here, as the loop seems too complex